src/client/App.tsx

```diff
@@ -62,24 +101,38 @@ class App extends React.Component {
     this.setState({ address: ev.currentTarget.value })
   }
 
+  private handleSelect: React.ChangeEventHandler<HTMLSelectElement> = ev => {
+    const { value: currency } = ev.currentTarget
+    this.setState(({ connections }) => ({
+      selectedConnection: connections.find(conn => conn.currency === currency)
+    }))
+  }
+
   private handleSubmit: React.FormEventHandler = async ev => {
     ev.preventDefault()
+
     this.setState({ pending: true, error: '' })
-    const response = await fetch(`/fund/${this.state.address}`, {
-      method: 'POST',
+
+    const { address, selectedConnection } = this.state
+
+    const response = await fetch(`/fund/${address}`, {
+      body: JSON.stringify(selectedConnection),
```

Comment:
I really don't think we should pass the entire connection from the client up to the server. I would only have the currency key as the selector. Choose the connections from the server side.